webapp/lib/router.dart

```diff
@@ -32,25 +43,31 @@ class Router {
   }
 
   void listen() {
-    window.onPopState.listen((PopStateEvent event) => _loadView(window.location.hash));
+    window.onPopState.listen((_) => _loadView(window.location.hash));
   }
 
   void routeTo(String path) {
     _loadView(path);
   }
 
+  Map<String, String> get routeParams => _currentRoute.params;
+
   void _loadView(String path) {
-    var targetRoute = _routes[path];
+    var pathParts = path.split('?');
+    _currentRoute = _routes[pathParts[0]];
```

Comment:
I think best to use the `Uri` class methods for this kind of URL processing

Reply:
and I think this should be moved down right before the `if (pathParts.length > 1) ` line, otherwise if we fall through one of the other `if`s we can end up in some pretty weird cases where, for example, the auth route has path parts from a project if the user isn't authenticated anymore..

Reply:
sorted out with your refactor